text/0000-use-case-integration-best-practices.md

```diff
@@ -0,0 +1,177 @@
+
+### Validate new transactions on more than one full node before accepting them
+
+When running more than one node, any new transaction must be accepted only after validating that it's a valid transaction in all the nodes.
+
+### Validate all your full nodes have the same best block
+
+It's also important to regularly check if all nodes have the same best block. If they have different best blocks, this might be a clue that one of the nodes could be a target of an attack because is running a separate fork of the blockchain.
+
+### Recommended architecture
+
+<pre>
+                       ┌─────────────┐      ┌──────────┐                     ┌─────────────┐        ┌─────────────┐
+                       │             │      │          │◀────────────────────┤             │        │             │
+                       │  Hathor     │      │ Headless │                     │  Use Case   │        │  Customer   │
+                       │ Full Node 1 │◀────▶│  Wallet  │                     │  Software   │◀──────▶│             │
+                       │             │      │          ├────────────────────▶│             │        │             │
+                       └────┬────────┘      └──────────┘                     └─────────────┘        └─────────────┘
+                            │
+┌────────────┐              │
+│            │◀─────────────┘
+│  Rest of   │
+│  Network   │
+│            │◀─────────────┐
+└────────────┘              │
+                       ┌────┴────────┐
+                       │             │               
+                       │  Hathor     │
+                       │ Full Node 2 │
+                       │             │
+                       └─────────────┘
+
+</pre>
+
+## Peer-id
+
+The peer-id is a unique identifier of your full node in Hathor's p2p network. Even though it does not contain any private key information, keeping it secret is a security measure that prevents attackers from targeting directly your full nodes.
+
+Because of that, it's important to never share it in public chats, only in private groups with the Hathor team. If you think your peer-id has been exposed, you should generate a new peer-id and replace the exposed ones.
+
+## Be alert for weird behavior
+
+### Correctly validate transactions
+
+The transactions in the Hathor network have many fields that must be checked to guarantee that a transaction is valid for your use case. For more details about the fields of a transaction, check the [Transaction Anatomy RFC](https://github.com/HathorNetwork/rfcs/blob/master/text/0015-anatomy-of-tx.md).
+
+#### Version
+
+Transactions have `version=1`, blocks have versions `0` or `3`, and token creation transactions have `version=2`.
```

Comment:
I feel we should explain it better. Should use cases verify version? How?

Reply:
That's one thing that I didn't like in this RFC, it doesn't describe in details the integration, so the answer for your "How" is: depends. 

If they are using the headless, is one way, if they are using the full node directly, is another way.

I could mention the exchange integration RFC (https://github.com/HathorNetwork/rfcs/blob/master/text/0027-exchange-integration.md#how-to-check-if-a-transaction-is-a-valid-deposit) and remove the whole "How to validate a transaction" section here, there we have all the API details. 

What do you think?

Reply:
I feel it should be a comprehensive document (instead of pointing to other documents). So I feel we should explain what should be checked. For example:

"Version identifies the type of vertex. Here are the choices: ... Depending on your use case, you might have different validation rules for each type."

"Voided state: A void transaction is cancelled and should never be accepted. You must validate that the transaction is not voided asserting that `is_voided == false`.

"Outputs: Hathor supports multi-token transactions. You must confirm that your outputs are correctly placed as follows:

* Token id is correct. If you accept only HTR token, token id must be `"00"`.
* Token data is equal to `0`.
* Value matches the deposit value. Note that it is an integer and `12.34` is represented by `1234`.
* Timelock must be null; otherwise your funds might be locked.
"

Reply:
I understand your idea but my main concern is that the fields depend on the integration. If the use case is getting information from the headless or the full node, the tx fields might be different.

But the text is generic now, changed to your suggestion. What do you think?

Reply:
Great job! I think it's way better. Thanks!